assets/queries-handler.js

```diff
@@ -53,19 +56,60 @@ if (searchTitle) {
   searchTitle.innerHTML = query;
 }
 
-if (sortSelect) {
+// dropdown select
+if (dropdown) {
   let sortField = urlParams.get('sort_field');
-  
+  let sortOrder = urlParams.get('sort_order');
+
   if (!sortField) {
     // Set a default value for sortField if it is null or undefined
-    sortField = sortSelect.options[0].value;
+    sortField = 'price';
+  }
+
+  if (!sortOrder) {
+    // Set a default value for sortOrder if it is null or undefined
+    sortOrder = 'asc';
   }
 
-  sortSelect.value = sortField;
-  
-  sortSelect.addEventListener('change', () => {
-    const convertedSortOrder = sortOrder === 'asc' ? 'desc' : 'asc';
-    const selectedValue = sortSelect.value;
-    window.location.href = convertUrlWithMultipleQuery(['sort_field', 'sort_order'], [selectedValue, convertedSortOrder]);
+  // Get the selected option from the URL parameters
+  const selectedOption = dropdownContent.querySelector(`[data-value="${sortField}-${sortOrder}"]`);
+
+  // Update the dropdown button text with the selected option
+  dropdownBtn.innerHTML = `<span class='order-by'>${order_by} : </span> ${selectedOption.textContent}`;
+
+  // Create a chevron icon for the dropdown button
+  const icon = document.createElement('ion-icon');
+  icon.setAttribute('name', 'chevron-down-outline');
+  icon.classList.add('dropdown-icon');
+  dropdownBtn.appendChild(icon);
+
+  // Set the font weight for the selected option
+  selectedOption.style.fontWeight = 'bold';
+
+  // Handle the click event for the dropdown button
+  dropdownBtn.addEventListener('click', () => {
+    dropdownContent.classList.toggle('show');
+  });
+
+  // Handle the click event for the options in the dropdown
+  dropdownContent.addEventListener('click', (event) => {
+    event.preventDefault();
+    const selectedValue = event.target.getAttribute('data-value');
+    const [newSortField, newSortOrder] = selectedValue.split('-');
+
+    if (newSortField !== sortField) {
+      // If a new sort field is selected, set the sort order to the default value
+      window.location.href = convertUrlWithMultipleQuery(['sort_field', 'sort_order'], [newSortField, 'asc']);
+    } else if (newSortOrder !== sortOrder) {
+      // If the same sort field is selected but with a different sort order, update the sort order
+      window.location.href = convertUrlWithMultipleQuery(['sort_field', 'sort_order'], [sortField, newSortOrder]);
+    }
+  });
```

Comment:
The dropdown component will be used elsewhere, such as in the product variant, it would be preferable if you could separate its logic from the query handler so that you may utilize it separately afterwards